### PR TITLE
User can download one or more tracks.

### PR DESCRIPTION
**Pull request by danielduffield**
Created a new view class to be displayed once the tracklist is submitted, revealing a table containing the tracklist information with individual download links, a button to download all tracks in a compressed folder, and a button to return to the root directory to submit a new url.

### CODE REVIEW

**Comment by danielduffield** [comment]
screen shot 2017-07-18 at 7 28 00 pm

File: client/main.js [hunk 1]
```diff
@@ -56,5 +61,20 @@ $tracklistForm.addEventListener('submit', event => {
   const tracklistPost = {}
   tracklistPost.tracklist = tracklist
   tracklistPost.metaData = albumMetadata
-  sendTracklistPostRequest(tracklistPost)
+  sendTracklistPostRequest(tracklistPost).then(zipPath => {
+    const $tracklistLinks = getTracklistLinks(tracklist, albumMetadata.videoId)
+    console.log($tracklistLinks)
```
Comment (victormejia) [inline client/main.js]: remove this log

File: client/utils/sendTracklistPostRequest.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
     method: 'POST',
     headers: {
       'Content-Type': 'application/json',
       'Accept': 'application/json'
     },
     body: JSON.stringify(tracklistWithData)
   })
-  .then(response => console.log(response))
+  .then(response => {
+    console.log(response)
```
Comment (victormejia) [inline client/utils/sendTracklistPostRequest.js]: remove this log

File: client/utils/sendTracklistPostRequest.js [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+    return response.json()
+  })
+  .then(json => {
+    console.log(json)
```
Comment (victormejia) [inline client/utils/sendTracklistPostRequest.js]: remove this log

File: client/utils/sendUrlPostRequest.js [hunk 4]
```diff
@@ -34,4 +33,23 @@ function transitionToTracklistForm(keyData) {
...
 }
 
+function resetTracklist() {
+  const $trackFormContainer = document.getElementById('track-form-container')
+  let resetIndex = 1
+  let $trackForm = document.querySelector('.track-form-' + resetIndex)
+  while ($trackForm) {
+    $trackFormContainer.removeChild($trackForm)
+    resetIndex++
+    $trackForm = document.querySelector('.track-form-' + resetIndex)
```
Comment (victormejia) [inline client/utils/sendUrlPostRequest.js]: is there a need to do this? vs. `$trackFormContainer.innerHTML = ''`

File: server/utils/compressTracklist.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      console.log('archiver has been finalized and the output file descriptor has closed.')
+      resolve('/download/' + youtubeId + '/tracklist.zip')
+    })
+
+    archive.on('warning', function (err) {
+      if (err.code === 'ENOENT') {
+        console.log('WARNING ERROR: ', err)
+      }
+      else {
+        throw err
```
Comment (victormejia) [inline server/utils/compressTracklist.js]: instead of throwing error here, reject the response

File: server/utils/downloadAlbum.js [hunk 6]
```diff
@@ -3,9 +3,13 @@ const fs = require('fs-extra')
 const path = require('path')
 
 function downloadAlbum(url, keyData) {
-  const albumFolder = path.join(__dirname, '/../../downloaded/' + keyData.videoId)
```
Comment (victormejia) [inline server/utils/downloadAlbum.js]: should downloadAlbum return a promise?

**Pull request merged**